### PR TITLE
added solution: Configure Spring MVC, created UserResponseDto, implem…

### PR DESCRIPTION
**Pull request by viaviaj**
added solution: 
-Configure Spring MVC, 
-created UserResponseDto, 
-implemted UserController.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,7 +52,35 @@
             <artifactId>mysql-connector-java</artifactId>
             <version>8.0.21</version>
         </dependency>
+
+        <dependency>
+            <groupId>javax.servlet</groupId>
```
Comment (jah701) [inline pom.xml:57]: I'm not sure if we have to keep this dependencies (jsp, jstp, servlets)

File: src/main/java/spring/intro/dao/UserDaoImpl.java [hunk 2]
```diff
@@ -38,6 +39,19 @@ public void add(User user) {
         }
     }
 
+    @Override
+    public Optional<User> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<User> query = session.createQuery("from User "
+                    + "where id = :id", User.class);
+            query.setParameter("id", id);
+            return query.uniqueResultOptional();
```
Comment (jah701) [inline src/main/java/spring/intro/dao/UserDaoImpl.java]: It's only a suggestion to use CriteriaAPI.
I suppose it works faster for select queries. 🤔

**Review by jah701 — APPROVED** [review]
Well done!

File: pom.xml (hunk 1 above)
Comment (viaviaj) [reply pom.xml:57]: It's dependency javax.servlet-api, we need it for work with Servlets AbstractAnnotationConfigDispatcherServletInitializer

File: src/main/java/spring/intro/dao/UserDaoImpl.java (hunk 2 above)
Comment (viaviaj) [reply src/main/java/spring/intro/dao/UserDaoImpl.java]: yes, it's optional, thanks

File: src/main/java/spring/intro/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import spring.intro.dto.UserResponseDto;
+import spring.intro.model.User;
+import spring.intro.service.UserService;
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+    AnnotationConfigApplicationContext context =
+            new AnnotationConfigApplicationContext(AppConfig.class);
+    UserService userService = context.getBean(UserService.class);
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/controllers/UserController.java]: why not autowiring?

File: src/main/java/spring/intro/controllers/UserController.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    @GetMapping
+    public List<UserResponseDto> getAll() {
+        return userService.listUsers().stream()
+                .map(this::mapUserResponseDto)
+                .collect(Collectors.toList());
+    }
+
+    private UserResponseDto mapUserResponseDto(User user) {
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/controllers/UserController.java]:
```suggestion
    private UserResponseDto mapToResponseDto(User user) {
```

File: src/main/java/spring/intro/dao/UserDaoImpl.java [hunk 5]
```diff
@@ -38,6 +39,19 @@ public void add(User user) {
         }
     }
 
+    @Override
+    public Optional<User> findById(Long id) {
+        try (Session session = sessionFactory.openSession()) {
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/dao/UserDaoImpl.java:43]: let's use session get instead that will be much simpler

File: src/main/java/spring/intro/controllers/UserController.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+@RequestMapping("/user")
+public class UserController {
+    private final UserService userService;
+
+    @Autowired
+    public UserController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping(value = "/inject")
```
Comment (RomanDubovskyi) [inline src/main/java/spring/intro/controllers/UserController.java]:
```suggestion
    @GetMapping("/inject")
```

**Review by RomanDubovskyi — APPROVED** [review]

**Comment by RomanDubovskyi** [comment]
travis failed, fix it before merge

**Pull request merged**